### PR TITLE
Misc updates

### PR DESCRIPTION
**Pull request by starkers**
# Breaking changes

- no longer support v1beta1 ingress objects
- only support network/v1ingresses

# bug fixes
- display the `ingressClassName` on the `routes` tab (UI)
- sort by deployment ages in descending order (most recently updated stuff at the top)

### CODE REVIEW

**Review by Lastin — APPROVED** [review]

**Review by anderson4u2 — APPROVED** [review]

File: frontend/src/components/TableDeployments.vue [hunk 1]
```diff
@@ -106,11 +110,43 @@ export default {
...
+      if (seconds < 60) {
+        return seconds + 's'
+      } else if (minutes < 60) {
+        return minutes + 'm'
+      } else if (hours < 24) {
+        return hours + 'h'
+      } else {
+        return days + 'd'
+      }
+    },
```
Comment (DanielPritchard) [inline frontend/src/components/TableDeployments.vue:148]: There is a lib for this that cleans it all up nicely: https://www.npmjs.com/package/pretty-ms

Comment (starkers) [reply frontend/src/components/TableDeployments.vue:148]: that'd be much nicer, will have a bash at it

Comment (starkers) [reply frontend/src/components/TableDeployments.vue:148]: Didn't have much luck loading this in vuejs, this function is reused by several components however so I'll start a util package for it in meantime

**Comment by starkers** [comment]
actually this branch went a little too far, (I got a little carried away in my free time with rewriting the frontend)

I'm going to clean it up so the PR is still appropriate to its original scope

**Comment by starkers** [comment]
I've added some GH workflows to publish the docker image here on github + run some linting

Also a fix to the Deployment page sorting being lost when data is loaded under the hood.

**Pull request merged**